### PR TITLE
[Clang] Fix LibTooling doc

### PR DESCRIPTION
**Pull request by maxmosk**
Replace CommonOptionsParser ctor by factory method ::create.

Found on page https://clang.llvm.org/docs/LibTooling.html

### CODE REVIEW

File: clang/docs/LibTooling.rst [hunk 1]
```diff
@@ -71,9 +71,9 @@ and automatic location of the compilation database using source files paths.
   int main(int argc, const char **argv) {
     // CommonOptionsParser constructor will parse arguments and create a
     // CompilationDatabase.  In case of error it will terminate the program.
-    CommonOptionsParser OptionsParser(argc, argv, MyToolCategory);
+    auto OptionsParser = CommonOptionsParser::create(argc, argv, MyToolCategory);
```
[the remark this answers is not in the document]
Comment (Sirraide) [reply clang/docs/LibTooling.rst:74]: Also, `->` needs to be replaced with `.` again below.

File: clang/docs/LibTooling.rst [hunk 2]
```diff
@@ -133,9 +133,9 @@ version of this example tool is also checked into the clang tree at
   static cl::extrahelp MoreHelp("\nMore help text...\n");
 
   int main(int argc, const char **argv) {
-    CommonOptionsParser OptionsParser(argc, argv, MyToolCategory);
-    ClangTool Tool(OptionsParser.getCompilations(),
-                   OptionsParser.getSourcePathList());
+    auto OptionsParser = CommonOptionsParser::create(argc, argv, MyToolCategory);
```
Comment (Sirraide) [inline clang/docs/LibTooling.rst:136]: As above

File: clang/include/clang/Tooling/CommonOptionsParser.h [hunk 3]
```diff
@@ -56,9 +56,9 @@ namespace tooling {
 /// ...
 ///
 /// int main(int argc, const char **argv) {
-///   CommonOptionsParser OptionsParser(argc, argv, MyToolCategory);
-///   ClangTool Tool(OptionsParser.getCompilations(),
-///                  OptionsParser.getSourcePathList());
+///   auto OptionsParser = CommonOptionsParser::create(argc, argv, MyToolCategory);
```
Comment (Sirraide) [inline clang/include/clang/Tooling/CommonOptionsParser.h:59]: And here too

**Review by Sirraide — CHANGES_REQUESTED** [review]
Thanks for improving our documentation!

There’s some error handling missing here. The `CommonOptionsParser` constructor would previously just exit, but that is no longer the case here, so we actually need to handle errors now.

**Comment by Sirraide** [comment]
I also just noticed that there already is a pr for this: #70427.

**Comment by maxmosk** [comment]
> I also just noticed that there already is a pr for this: #70427.

This PR wasn't approved and doc is still not fixed :(
What to do with this PR?

**Comment by Sirraide** [comment]
> This PR wasn't approved and doc is still not fixed :( What to do with this PR?

It somehow just went unnoticed all this time, unfortunately. I’ll approve the other pr since it seems to do the same thing that we’re also doing somewhere else in our documentation, and it’s pretty much what I suggested here anyway.

The comment in `clang/Tooling/CommonOptionsParser.h` still needs to be updated, which the other pr doesn’t do; you can still do that if you’d like. Sorry about the entire confusion here; this is something that shouldn’t normally happen...

**Comment by maxmosk** [comment]
@Sirraide updated with header doc fix, please review

**Review by Sirraide — APPROVED** [review]
LGTM seeing as this matches what’s already in the documentation.

**Comment by Sirraide** [comment]
Right, what `clang-format` is suggesting here is a bit bizzare; please reformat the comment manually so every line fits within the 80 column limit.

**Review by Sirraide — APPROVED** [review]

**Pull request merged**